cpp/src/io/comp/common.cpp

```diff
@@ -54,4 +56,32 @@ namespace cudf::io::detail {
   }
 }
 
+[[nodiscard]] size_t find_split_index(device_span<device_span<uint8_t const> const> inputs,
+                                      host_engine_state host_state,
+                                      size_t auto_mode_threshold,
+                                      size_t hybrid_mode_target_ratio,
+                                      rmm::cuda_stream_view stream)
+{
+  CUDF_FUNC_RANGE();
+  if (host_state == host_engine_state::OFF or inputs.empty()) { return 0; }
+  if (host_state == host_engine_state::ON) { return inputs.size(); }
+
+  if (host_state == host_engine_state::AUTO) {
+    return inputs.size() < auto_mode_threshold ? inputs.size() : 0;
+  }
+
+  if (host_state == host_engine_state::HYBRID) {
+    auto const h_inputs    = cudf::detail::make_host_vector(inputs, stream);
+    size_t total_host_size = 0;
+    for (size_t i = 0; i < h_inputs.size(); ++i) {
+      if (total_host_size >= hybrid_mode_target_ratio * h_inputs[i].size()) { return i; }
```

Comment:
Maybe I didn't understand this correctly but couldn't this lead to false positives results for skewed input sizes? Do we instead want to do a partial sum and then find the split (based on target ratio). Also, do we need to check if target ratio is within a certain range?

Reply:
Oh wait, these are sorted, right?

Reply:
Thanks. This makes sense then.